unyt/_array_functions.py

```diff
@@ -701,3 +709,13 @@ def cumprod(a, *args, **kwargs):
         "with a unyt_array as all return elements should (but cannot) "
         "have different units."
     )
+
+
+@implements(np.pad)
+def pad(array, *args, **kwargs):
+    return np.pad._implementation(array.view(np.ndarray), *args, **kwargs) * array.units
+
+
+@implements(np.sinc)
+def sinc(x, *args, **kwargs):
+    return np.sinc._implementation(x.view(np.ndarray), *args, **kwargs) * x.units
```

Comment:
Shouldn't this have units of `1/x.units`?

Reply:
It might also make sense to return a dimensionless number and ignoring the input units like we do for other transcendental functions.

FWIW, here's what astropy does:
<details>

```
In [3]: np.sinc([1, 2]*gram)
---------------------------------------------------------------------------
UnitConversionError                       Traceback (most recent call last)
File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/quantity.py:899, in Quantity.to_value(self, unit, equivalencies)
    898 try:
--> 899     scale = self.unit._to(unit)
    900 except Exception:
    901     # Short-cut failed; try default (maybe equivalencies help).

File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/core.py:1093, in UnitBase._to(self, other)
   1091         return self_decomposed.scale / other_decomposed.scale
-> 1093 raise UnitConversionError(
   1094     f"'{self!r}' is not a scaled version of '{other!r}'")

UnitConversionError: 'Unit("g")' is not a scaled version of 'Unit("rad")'

During handling of the above exception, another exception occurred:

UnitConversionError                       Traceback (most recent call last)
File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/quantity_helper/function_helpers.py:219, in sinc(x)
    218 try:
--> 219     x = x.to_value(radian)
    220 except UnitsError:

File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/quantity.py:902, in Quantity.to_value(self, unit, equivalencies)
    900 except Exception:
    901     # Short-cut failed; try default (maybe equivalencies help).
--> 902     value = self._to_value(unit, equivalencies)
    903 else:

File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/quantity.py:809, in Quantity._to_value(self, unit, equivalencies)
    807 if not self.dtype.names or isinstance(self.unit, StructuredUnit):
    808     # Standard path, let unit to do work.
--> 809     return self.unit.to(unit, self.view(np.ndarray),
    810                         equivalencies=equivalencies)
    812 else:
    813     # The .to() method of a simple unit cannot convert a structured
    814     # dtype, so we work around it, by recursing.
    815     # TODO: deprecate this?
    816     # Convert simple to Structured on initialization?

File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/core.py:1130, in UnitBase.to(self, other, value, equivalencies)
   1129 else:
-> 1130     return self._get_converter(Unit(other),
   1131                                equivalencies=equivalencies)(value)

File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/core.py:1061, in UnitBase._get_converter(self, other, equivalencies)
   1059                 pass
-> 1061 raise exc

File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/core.py:1046, in UnitBase._get_converter(self, other, equivalencies)
   1045 try:
-> 1046     return self._apply_equivalencies(
   1047         self, other, self._normalize_equivalencies(equivalencies))
   1048 except UnitsError as exc:
   1049     # Last hope: maybe other knows how to do it?
   1050     # We assume the equivalencies have the unit itself as first item.
   1051     # TODO: maybe better for other to have a `_back_converter` method?

File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/core.py:1022, in UnitBase._apply_equivalencies(self, unit, other, equivalencies)
   1020 other_str = get_err_str(other)
-> 1022 raise UnitConversionError(
   1023     f"{unit_str} and {other_str} are not convertible")

UnitConversionError: 'g' (mass) and 'rad' (angle) are not convertible

During handling of the above exception, another exception occurred:

UnitTypeError                             Traceback (most recent call last)
Cell In [3], line 1
----> 1 np.sinc([1, 2]*gram)

File <__array_function__ internals>:185, in sinc(*args, **kwargs)

File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/quantity.py:1720, in Quantity.__array_function__(self, function, types, args, kwargs)
   1718 function_helper = FUNCTION_HELPERS[function]
   1719 try:
-> 1720     args, kwargs, unit, out = function_helper(*args, **kwargs)
   1721 except NotImplementedError:
   1722     return self._not_implemented_or_raise(function, types)

File ~/.pyenv/versions/3.10.8-debug/lib/python3.10/site-packages/astropy/units/quantity_helper/function_helpers.py:221, in sinc(x)
    219     x = x.to_value(radian)
    220 except UnitsError:
--> 221     raise UnitTypeError("Can only apply 'sinc' function to "
    222                         "quantities with angle units")
    223 return (x,), {}, dimensionless_unscaled, None

UnitTypeError: Can only apply 'sinc' function to quantities with angle units
```

</details>

We aren't as restrictive as astropy about requiring angle units for these transcendental functions, so I'd probably just make this behavior exactly mirror how we handle the `np.sin` ufunc: just ignore the units and return a dimensionless array.

Reply:
oh, nice catch, and thank you for these pointers.
> I'd probably just make this behavior exactly mirror how we handle the np.sin ufunc: just ignore the units and return a dimensionless array.

I just pushed this change.